flatten_array.rb

```diff
@@ -0,0 +1,28 @@
+require 'pry'
+
+class FlattenArray
+  def self.flatten(array)
+    flattened_array = []
+
+    array.each do |element|
+      flattened_array = self.add_element_array(element, flattened_array)
+    end
+    flattened_array.compact
```

Comment:
Add a ! after compact to modify the original array.

Reply:
If we added a `!` it would return nil if no changes were made, which we might want to avoid